lib/mobility-core/src/Kernel/Storage/Clickhouse/Queries.hs

```diff
@@ -42,37 +50,39 @@ __orderBy = addClause
 __offset :: String -> Offset -> String
 __offset = addClause
 
-runClickhouse :: (MonadFlow m, ClickhouseFlow m env, FromJSON a) => (HttpConnection -> IO (Either String a)) -> m (Either String a)
-runClickhouse action = do
-  con <- asks (.clickhouseEnv.connection)
+runClickhouse :: (MonadFlow m, ClickhouseFlow m env, FromJSON a) => (HttpConnection -> IO (Either String a)) -> ClickhouseDb -> m (Either String a)
+runClickhouse action db = do
+  con <- case db of
+    ATLAS_DRIVER_OFFER_BPP -> asks (.driverClickhouseEnv.connection)
+    ATLAS_KAFKA -> asks (.kafkaClickhouseEnv.connection)
   L.runIO $ action con
 
-runRawQuery :: (MonadFlow m, ClickhouseFlow m env, FromJSON a) => String -> m (Either String a)
+runRawQuery :: (MonadFlow m, ClickhouseFlow m env, FromJSON a) => String -> ClickhouseDb -> m (Either String a)
 runRawQuery query = runClickhouse (`runQuery` getJSON query)
 
-runRawQuery' :: (MonadFlow m, ClickhouseFlow m env, FromJSON a) => ClickhouseExpr -> m (Either String a)
-runRawQuery' (ExprStr query) = runRawQuery query
-runRawQuery' _ = throwError $ InternalError "can't call this function with unresolved clickhouseExpr"
+runRawQuery' :: (MonadFlow m, ClickhouseFlow m env, FromJSON a) => ClickhouseExpr -> ClickhouseDb -> m (Either String a)
+runRawQuery' (ExprStr query) db = runRawQuery query db
+runRawQuery' _ _ = throwError $ InternalError "can't call this function with unresolved clickhouseExpr"
 
-constructQuery :: Typeable a => Proxy a -> ClickhouseExpr -> Maybe Limit -> Maybe Offset -> Maybe Order -> String
-constructQuery proxyTable expr mbLimit mbOffset mbOrder = do
-  addOffset . addLimit . addOrder $ __select proxyTable expr
+constructQuery :: Typeable a => Proxy a -> [Text] -> Maybe String -> ClickhouseExpr -> Maybe Limit -> Maybe Offset -> Maybe Order -> String
+constructQuery proxyTable columns maybeGroupBy expr mbLimit mbOffset mbOrder = do
+  addOffset . addLimit . addOrder $ __select proxyTable columns maybeGroupBy expr
```

Comment:
Lets add appendGroup here link other maybe things i.e. addLimit etc. any non maybe thing we can pass to the `__select` function.


Also as we are touching here, can you make all the non maybe things come in end of the function call ? Order wise Left -> Right: Maybes -> List -> NonEmpty List -> Non Maybe values. This way it will help cleaner to create specialized function like findOneWhere, findOneWhereWithLimit etc. and let's create these functions as well to make NY look cleaner.